src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -343,28 +343,29 @@ public void TestCreateNoWindowProperty(bool value)
         }
 
         [Fact]
-        public void TestWorkingDirectoryProperty()
+        public void TestWorkingDirectoryPropertyInChildProcess()
         {
             CreateDefaultProcess();
 
             // check defaults
             Assert.Equal(string.Empty, _process.StartInfo.WorkingDirectory);
 
-            Process p = CreateProcessLong();
-            p.StartInfo.WorkingDirectory = Directory.GetCurrentDirectory();
+            ProcessStartInfo psi = new ProcessStartInfo();
+            string sourcePath = Directory.GetCurrentDirectory();
 
-            try
-            {
-                p.Start();
-                Assert.Equal(Directory.GetCurrentDirectory(), p.StartInfo.WorkingDirectory);
-            }
-            finally
+            foreach (string newPath in Directory.GetFiles(sourcePath, "*.*", SearchOption.AllDirectories))
+                File.Copy(newPath, newPath.Replace(sourcePath, TestDirectory), true);
```

Comment:
Why do we need to copy files?

Reply:
Process.Start runs the executable named ```psi.filename``` in psi.workingDirectory```.
Here the executable file is RemoteExecutorConsoleApp.exe.
So in order to start this process. We need to have this this executable file in our new Working Directory.

Reply:
If you set psi.FileName to a full path it should no longer find the exe in psi.WorkingDirectory

Reply:
I tried this 
```
            ProcessStartInfo psi = new ProcessStartInfo();
            string sourcePath = Directory.GetCurrentDirectory();

            //foreach (string newPath in Directory.GetFiles(sourcePath, "*.*", SearchOption.AllDirectories))
            //    File.Copy(newPath, newPath.Replace(sourcePath, TestDirectory), true);

            psi.WorkingDirectory = TestDirectory;
            psi.FileName = @"C:\git\corefx\bin\tests\System.Diagnostics.Process.Tests\netcoreapp-Windows_NT-Debug-x64\RemoteExecutorConsoleApp.exe";
```
The Error says   No executable found matching command "dotnet-RemoteExecutorConsoleApp.exe"
     System.Diagnostics.Tests.ProcessStartInfoTests.TestWorkingDirectoryPropertyInChildProcess [FAIL]

Reply:
Hmm, dig through RemoteExecutorTestBase*cs, which is what this test is ultimately using to start the process.

I see 
```
protected static readonly string HostRunner = Process.GetCurrentProcess().MainModule.FileName; 
```
```
            if (!File.Exists(HostRunner))
                throw new IOException($"{HostRunner} test app isn't present in the test runtime directory.");
...
             psi.FileName = HostRunner;
```

It looks like this method ` private static RemoteInvokeHandle RemoteInvoke(MethodInfo method, string[] args, RemoteInvokeOptions options, bool pasteArguments = true)` overwrites options.psi.FileName entirely. If I'm reading right, you'll want it to not do that if it's set, I think. if you can let you put a full path in there, it should work.